Classes/common/touchdb/TDRemoteRequest.m

```diff
@@ -86,74 +98,75 @@ - (void)setAuthorizer:(id<TDAuthorizer>)authorizer
     }
 }
 
-- (void)setupRequest:(NSMutableURLRequest *)request withBody:(id)body
-{
-    // subclasses can override this.
-}
-
 - (void)dontLog404 { _dontLog404 = true; }
 
 - (void)start
 {
     if (!_request) return;  // -clearConnection already called
     CDTLogVerbose(CDTTD_REMOTE_REQUEST_CONTEXT, @"%@: Starting...", self);
-    Assert(!_connection);
-    _connection = [NSURLConnection connectionWithRequest:_request delegate:self];
-
-// Retaining myself shouldn't be necessary, because NSURLConnection is documented as retaining
-// its delegate while it's running. But GNUstep doesn't (currently) do this, so for
-// compatibility I retain myself until the connection completes (see -clearConnection.)
-// TODO: Remove this and the [self autorelease] below when I get the fix from GNUstep.
-#ifdef GNUSTEP
-    [self retain];
-#endif
+
+    __weak TDRemoteRequest *weakSelf = self;
+    self.task = [self.session dataTaskWithRequest:_request
+                                completionHandler:^(NSData *data,
+                                                    NSURLResponse *response,
+                                                    NSError *error) {
+        TDRemoteRequest * strongSelf = weakSelf;
+        if(error) {
+            [strongSelf requestDidError:error];
+        } else if (TDStatusIsError(((NSHTTPURLResponse *)response).statusCode)) {
+            [strongSelf receivedResponse:response];
+        }  else {
+            [strongSelf receivedResponse:response];
+            [strongSelf receivedData:data];
+        }
+    }];
+    [self.task resume];
+
 }
 
-- (void)clearConnection
+- (void)clearSession
 {
     _request = nil;
-    if (_connection) {
-        _connection = nil;
-
-#ifdef GNUSTEP
-        [self release];
-#endif
-    }
+    [self.task cancel];
+    self.task = nil;
+    //doesn't really do anything.
```

Comment:
Given it's the end of a method, this comment is trivially true. Was there something happening here before?